src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Tag} matches any of the keywords given.
+ */
+public class TagContainsKeywordsPredicate implements Predicate<Person> {
+    private final List<String> keywords;
+
+    public TagContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        int numTags = person.getTags().size();
+        boolean hasAllMatches = true;
+
+        for (int i = 0; i < keywords.size(); i++) {
+            String keyword = keywords.get(i);
+            boolean doesKeywordMatchAnyTags = person.getTags().stream()
+                    .anyMatch(tag -> StringUtil.containsWordIgnoreCase(tag.tagName, keyword));
+            hasAllMatches = hasAllMatches && doesKeywordMatchAnyTags;
+        }
```

Comment:
Might want to consdeir using a for-each loop here `for (String keyword : keywords)`.
I think it increases readability and abstraction (which makes it safer)!

Reply:
Yup I agree. Will change it! 👌